src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteFactory.cs

```diff
@@ -573,13 +640,19 @@ private static void VerifyOpenGenericAotCompatibility(Type serviceType, Type[] g
             }
         }
 
-        public void Add(Type type, ServiceCallSite serviceCallSite)
+        public void Add(ServiceIdentifier serviceIdentifier, ServiceCallSite serviceCallSite)
         {
-            _callSiteCache[new ServiceCacheKey(type, DefaultSlot)] = serviceCallSite;
+            _callSiteCache[new ServiceCacheKey(serviceIdentifier, DefaultSlot)] = serviceCallSite;
         }
 
-        public bool IsService(Type serviceType)
+        public bool IsService(Type serviceType) => IsService(new ServiceIdentifier(null, serviceType));
+
+        public bool IsKeyedService(Type serviceType, object key) => IsService(new ServiceIdentifier(key, serviceType));
```

Comment:
Are there any tests for this? You need to be able to resolve the IKeyedServiceProviderIsService from the container which means it needs to be added to the CallSiteFactory in the ServiceProvider ctor like IServiceProviderIsService. We should add a specification test for this.

Reply:
Added tests, and added IServiceProviderIsKeydService to the built-in services